frontend/src/lib/util/urls.ts

```diff
@@ -0,0 +1,19 @@
+import type { ConditionalPick } from 'type-fest';
+import type { PrimitiveRecord } from './types';
+
+// https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1568#issuecomment-1587963141
+export function toSearchParams(params: PrimitiveRecord): string {
+  const searchParams = new URLSearchParams(params as unknown as Record<string, string>);
+  return searchParams.toString();
+}
+
+export function getBoolSearchParam<T extends PrimitiveRecord>(key: keyof ConditionalPick<T, boolean | null> & string, params: URLSearchParams): boolean | null {
```

Comment:
can we make this return just a bool, never null. A default value can be set via another parameter but we can default to false. It keeps us from having 3 states in our code for true, false or null, when there should never be more than 2 if you're using a bool.